lib/formtastic/inputs/base/validations.rb

```diff
@@ -138,7 +138,11 @@ def required?
           if validations?
             validations.select { |validator|
               if validator.options.key?(:on)
-                return false if (validator.options[:on] != :save) && ((object.new_record? && validator.options[:on] != :create) || (!object.new_record? && validator.options[:on] != :update))
+                if validator.options[:on].is_a?(Array)
+                  return false if (validator.options[:on].exclude?(:save)) && ((object.new_record? && validator.options[:on].exclude?(:create)) || (!object.new_record? && validator.options[:on].exclude?(:update)))
```

Comment:
I think this can be more simple because you don't actually need that `if` statement, just return the opposite value will do the job
```ruby
!(validator.options[:on].exclude?(:save) &&
  ((object.new_record? && validator.options[:on].exclude?(:create)) ||
   (!object.new_record? && validator.options[:on].exclude?(:update))
))
```
And you can also create an private method so that the condition can be more readable
```ruby
def exclude_action?(action)
  validator.options[:on].exclude?(action)
end
```

```ruby
!(exclude_action?(:save) &&
  ((object.new_record? && exclude_action?(:create)) ||
   (!object.new_record? && exclude_action?(:update))
))
```

Just some advise :smile: